pico/tools/src/main/java/io/helidon/pico/tools/ModuleInfoDescriptor.java

```diff
@@ -0,0 +1,633 @@
+
+    /**
+     * Loads the {@code module-info} descriptor given its source file location and preferred ordering scheme.
+     *
+     * @param file      the source file location
+     * @param ordering  the ordering to apply
+     * @return the module-info descriptor
+     * @throws java.io.IOException if there is any exception encountered
+     */
+    static ModuleInfoDescriptor load(File file,
+                                     Ordering ordering) throws IOException {
+        try {
+            Path filePath = file.toPath();
+            String moduleInfo = Files.readString(filePath);
+            return load(moduleInfo, ordering);
+        } catch (ToolsException e) {
+            throw new ToolsException("unable to load: " + file, e);
+        }
+    }
+
+    /**
+     * Loads the {@code module-info} descriptor given its source input stream.
+     *
+     * @param is the source file input stream
+     * @return the descriptor
+     * @throws io.helidon.pico.tools.ToolsException if there is any exception encountered
+     */
+    static ModuleInfoDescriptor uncheckedLoad(InputStream is) {
+        try {
+            return load(is, Ordering.NATURAL_PRESERVE_COMMENTS);
+        } catch (IOException e) {
+            throw new ToolsException("unable to load", e);
+        }
+    }
+
+    /**
+     * Loads the {@code module-info} descriptor given its input stream.
+     *
+     * @param is the source file location
+     * @param ordering the ordering to apply
+     * @return the descriptor
+     * @throws java.io.IOException if there is any exception encountered
+     */
+    static ModuleInfoDescriptor load(InputStream is,
+                                     Ordering ordering) throws IOException {
+        String moduleInfo = new String(is.readAllBytes(), StandardCharsets.UTF_8);
+        return load(moduleInfo, ordering);
+    }
+
+    /**
+     * Loads the descriptor given its literal source.
+     *
+     * @param moduleInfo the source
+     * @return the descriptor
+     * @throws io.helidon.pico.tools.ToolsException if there is any exception encountered
+     */
+    static ModuleInfoDescriptor uncheckedLoad(String moduleInfo) {
+        try {
+            return load(moduleInfo, Ordering.NATURAL_PRESERVE_COMMENTS);
+        } catch (IOException e) {
+            throw new ToolsException("unable to load", e);
+        }
+    }
+
+    /**
+     * Loads the descriptor given its literal source.
+     *
+     * @param moduleInfo    the source
+     * @param ordering      the ordering to apply
+     * @return the descriptor
+     * @throws java.io.IOException if there is any IO exceptions encountered
+     * @throws io.helidon.pico.tools.ToolsException if there is any other exception encountered
+     */
+    static ModuleInfoDescriptor load(String moduleInfo,
+                                     Ordering ordering) throws IOException {
+        final DefaultModuleInfoDescriptor.Builder descriptor = DefaultModuleInfoDescriptor.builder();
```

Comment:
Do not use `final` except for fields and classes (and methods if it makes sense)

Reply:
next push